graphql/redact_secrets_plugin.go

```diff
@@ -0,0 +1,117 @@
+package graphql
+
+import (
+	"fmt"
+	"os"
+	"sort"
+
+	"github.com/99designs/gqlgen/codegen/config"
+)
+
+// GenerateSecretFields generates a file that contains a list of fields that should be redacted in logs.
+func GenerateSecretFields(cfg *config.Config) error {
+	redactedFields := make(map[string]bool)
+
+	for _, schemaType := range cfg.Schema.Types {
+		for _, field := range schemaType.Fields {
+			if field.Directives.ForName("redactSecrets") != nil {
+				redactedFields[field.Name] = true
+			}
+		}
+	}
+
+	var fields []string
+	for field := range redactedFields {
+		fields = append(fields, field)
+	}
+
+	// Sort the fields to ensure consistent output.
+	sort.Sort(sort.StringSlice(fields))
+
+	return generateRedactedFieldsFile(fields)
+}
+
+// generateRedactedFieldsFile generates a file that contains a list of fields that should be redacted in logs. It is used to generate the redacted_fields_gen.go file.
+// The file contains a list of fields that should be redacted in logs, a function to check if a field should be redacted, and a function to redact fields in a map.
+func generateRedactedFieldsFile(fields []string) error {
```

Comment:
Instead of generating a file with all the code, would it be possible to just create a file (can be really simple, e.g., newline or comma separated field names) that is read in by this code here? This can be done with an `init` func that that writes the values to the global `RedactedFields` slice.

This will make the code here much easier to understand and test.

Reply:
This process is a bit intensive to actually generate the file since it involves analyzing the entire graphql schema. It takes about 14 seconds to run the script (`generate_secret_fields.go`). So I think adding that to the start up script everytime someone wants to start the local evergreen server or deploy would be a bit intense.

Reply:
This implementation only requires running the script once when someone make a change to the graphql schema and the value is loaded into memory automatically. 
The rest of the graphql code is codegeneration so this script is just an extension of that.

Reply:
Right, but you can instead run the script that analyzes the entire graphql schema once and write the fields to a file, upon startup the `init` func would only read in those fields to the `RedactedFields` slice (not run the entire script). Does that make sense?